virtcontainers/network.go

```diff
@@ -1378,12 +1394,15 @@ func createEndpoint(netInfo NetworkInfo, idx int, model NetInterworkingModel) (E
 		} else if netInfo.Iface.Type == "tap" {
 			networkLogger().Info("tap interface found")
 			endpoint, err = createTapNetworkEndpoint(idx, netInfo.Iface.Name)
+		} else if netInfo.Iface.Type == "tuntap" {
```

Comment:
You also need to check the Mode for the interface to determine is this is a `tap` or `tun` device:
https://github.com/vishvananda/netlink/blob/df0136952828fc27513ab7b03b3423e80f53e8cb/link.go#L310
Note: if you are seeing the Mode as 0, then it means the Mode was not determined correctly.
Reason being you are running a kernel < 4.13, and so the Mode was not determined through netlink. In that case, you may need to mount /sys.

Reply:
I'm a little confused about how the `Mode` will determine whether its a `tap` or `tun`, and how to get the `Mode` since all we have is the `NetworkInfo`. I will add a check for if the mode is 0, and then try mounting `/sys/` for /`sys/class/net`

Reply:
@amshinde can you clarify this further?

Reply:
@amshinde I ran on kernel 4.12 fedora 27, and it did not require the mode to get the links being passed. I haven't found a moment when mode 0 is being reached. Would you still like me to add the logic to mount /sys if that occurs?

Reply:
I should mention that the Mode is 0 for the older kernels, but if I don't check the mode and instead check the type it works by creating the tuntap endpoint as designed.

Reply:
@gabibeyer The Link.Type() is just going to tell you that this is a Tuntap device, the mode for the Tuntap struct tells you specifically whether this is a tun device or tap. We need to make that distinction.
I had tried this on 4.13.0 kernel  on Ubuntu where netlink did not give the information.
In that case, it is necessary to get that information from sysfs. However, it is the upper layer that should be mounting this instead of kata.
I know for sure rootless docker does this, is this is missing from podman that it should be added there.

Reply:
https://github.com/containers/libpod/pull/3831